client/src/components/BecomeChefButton.js

```diff
@@ -0,0 +1,62 @@
+import React, {useState, useContext} from "react";
+import Dialog from "common/Dialog"
+import Form from "common/Form";
+import {
+  Button
+} from "@material-ui/core";
+import { Context as MealContext } from "contexts/MealContext";
+import { Context as UserContext } from "contexts/AuthContext";
+
+
+const BecomeChefButton = ({loggedInUser}) => {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const { createMeal } = useContext(MealContext);
+  const { refreshUser } = useContext(UserContext)
+  const isChef = loggedInUser.isChef || false;
+  const fields = [
+    {
+      name: "mealName",
+      label: "Meal Name",
+      validation: {
+        required: "Name is required.",
+      },
+    },
+    {
+      name: "description",
+      label: "Description",
+    },
```

Comment:
I think there is a bit more involved in the fields for a meal (such as the number of people this is for, the price of the meal).